### PR TITLE
Add `Options` menu item that open our options page

### PR DESCRIPTION
**Pull request by rita-gorokhod**
Fixes [#3428](https://github.com/SonarSource/sonarlint-visualstudio/issues/3428)

image

### CODE REVIEW

File: src/Integration.Vsix.UnitTests/Commands/HelpMenu/AboutDialogTests.cs [hunk 1]
```diff
@@ -7,7 +27,7 @@
 namespace SonarLint.VisualStudio.Integration.UnitTests.Commands.HelpMenu
 {
     [TestClass]
-    internal class AboutDialogTests
+    public class AboutDialogTests
```
Comment (rita-sonarsource) [inline src/Integration.Vsix.UnitTests/Commands/HelpMenu/AboutDialogTests.cs:30]: Minor unrelated change, I noticed that there was no copyright + the class was internal (so resharper's unit tests runner doesn't recognize it)

File: src/Integration.Vsix/Commands/PackageCommandManager.cs [hunk 2]
```diff
@@ -31,41 +32,42 @@ namespace SonarLint.VisualStudio.Integration.Vsix
...
             IShowInBrowserService showInBrowserService,
-            IBrowserService browserService)
+            IBrowserService browserService,
+            ShowOptionsPage showOptionsPage)
         {
-            this.RegisterCommand((int)PackageCommandId.ProjectExcludePropertyToggle, new ProjectExcludePropertyToggleCommand(projectPropertyManager, projectToLanguageMapper));
-            this.RegisterCommand((int)PackageCommandId.ProjectTestPropertyAuto, new ProjectTestPropertySetCommand(projectPropertyManager, projectToLanguageMapper, null));
-            this.RegisterCommand((int)PackageCommandId.ProjectTestPropertyTrue, new ProjectTestPropertySetCommand(projectPropertyManager, projectToLanguageMapper, true));
-            this.RegisterCommand((int)PackageCommandId.ProjectTestPropertyFalse, new ProjectTestPropertySetCommand(projectPropertyManager, projectToLanguageMapper, false));
+            RegisterCommand((int)PackageCommandId.ProjectExcludePropertyToggle, new ProjectExcludePropertyToggleCommand(projectPropertyManager, projectToLanguageMapper));
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/PackageCommandManager.cs:52]: cosmetics: removed "this."

File: src/Integration.Vsix/Commands/PackageCommandManager.cs [hunk 3]
```diff
@@ -31,41 +32,42 @@ namespace SonarLint.VisualStudio.Integration.Vsix
...
+            RegisterCommand((int)PackageCommandId.ProjectTestPropertyAuto, new ProjectTestPropertySetCommand(projectPropertyManager, projectToLanguageMapper, null));
+            RegisterCommand((int)PackageCommandId.ProjectTestPropertyTrue, new ProjectTestPropertySetCommand(projectPropertyManager, projectToLanguageMapper, true));
+            RegisterCommand((int)PackageCommandId.ProjectTestPropertyFalse, new ProjectTestPropertySetCommand(projectPropertyManager, projectToLanguageMapper, false));
 
             // Menus
-            this.RegisterCommand((int)PackageCommandId.ProjectSonarLintMenu, new ProjectSonarLintMenuCommand(projectPropertyManager, projectToLanguageMapper));
+            RegisterCommand((int)PackageCommandId.ProjectSonarLintMenu, new ProjectSonarLintMenuCommand(projectPropertyManager, projectToLanguageMapper));
+
+            // Commands
+            RegisterCommand(CommonGuids.SonarLintMenuCommandSet, OptionsCommand.Id, new OptionsCommand(showOptionsPage));
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/PackageCommandManager.cs:61]: this is the new command

File: src/Integration.Vsix/SonarLint.vsct [hunk 4]
```diff
@@ -28,8 +28,13 @@
       <Group guid="SonarLintCmdSet" id="SonarLintOuterGroup" priority="0x0100">
         <Parent guid="guidSHLMainMenu" id="IDM_VS_MENU_EXTENSIONS" />
       </Group>
-      <!-- The sub menu that parents commands. -->
-      <Group guid="SonarLintCmdSet" id="SonarLintGroup" priority="0x0100">
+      <Group guid="SonarLintCmdSet" id="SonarLintConnectedModeGroup" priority="0x0100">
+        <Parent guid="SonarLintCmdSet" id="SonarLintMenu" />
+      </Group>
+      <Group guid="SonarLintCmdSet" id="SonarLintCommandsGroup" priority="0x0101">
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/SonarLint.vsct:34]: I renamed SonarLintGroup to SonarLintCommandsGroup and created 2 additional groups for help and connected mode; this is so I could have the dividers.

File: src/Integration.Vsix/SonarLint.vsct [hunk 5]
```diff
@@ -41,19 +46,30 @@
           <ButtonText>SonarLint</ButtonText>
         </Strings>
       </Menu>
-      <Menu guid="SonarLintCmdSet" id="HelpMenu" priority="0x9999">
-        <Parent guid="SonarLintCmdSet" id="SonarLintGroup"/>
+
+      <Menu guid="SonarLintCmdSet" id="ConnectedModeMenu">
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/SonarLint.vsct:50]: Cosmetics: Put them in the order they appear

File: src/Integration.Vsix/SonarLintIntegrationPackage.cs [hunk 6]
```diff
@@ -98,7 +98,9 @@ private async System.Threading.Tasks.Task InitOnUIThreadAsync()
                     serviceProvider.GetMefService<IProjectToLanguageMapper>(),
                     serviceProvider.GetMefService<IOutputWindowService>(),
                     serviceProvider.GetMefService<IShowInBrowserService>(),
-                    serviceProvider.GetMefService<IBrowserService>());
+                    serviceProvider.GetMefService<IBrowserService>(),
+                    ShowOptionPage);
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/SonarLintIntegrationPackage.cs:102]: ShowOptionPage is a method on this class, implemented by VS

File: src/Integration.Vsix.UnitTests/Commands/HelpMenu/AboutDialogTests.cs [hunk 7]
```diff
@@ -1,28 +0,0 @@
-﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
-using Moq;
-using SonarLint.VisualStudio.Integration.Vsix.Commands;
-using SonarLint.VisualStudio.Core;
-using System.Windows.Navigation;
-
-namespace SonarLint.VisualStudio.Integration.UnitTests.Commands.HelpMenu
-{
-    [TestClass]
-    internal class AboutDialogTests
```
Comment (rita-sonarsource) [inline src/Integration.Vsix.UnitTests/Commands/HelpMenu/AboutDialogTests.cs:10]: This class was Internal, so the build server wasn't actually picking it up. When I changed it to public, the build started to fail because we're creating a real AboutDialog and InitializeComponent is called and fails due to missing VS stuff.

Putting code directly in the xaml is a bad practice because of it but since it's one line, it's keep it as is for now and I deleted the test. If we want to test it, we need to take it out of the xaml.cs

File: src/Integration.Vsix/SonarLint.vsct (hunk 4 above)
Comment (bigfluffycookie) [reply src/Integration.Vsix/SonarLint.vsct:36]: Nice!

**Review by bigfluffycookie — APPROVED** [review]
Looks good!

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]
